ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java

```diff
@@ -326,6 +326,13 @@ private String sql(final String key) {
 
     @Override
     public void addTimer(final TimerImpl timerEntity) {
+        String timedObjectId = timerEntity.getTimedObjectId();
+        synchronized (this) {
+            if(!knownTimerIds.containsKey(timedObjectId)) {
+                EjbLogger.EJB3_TIMER_LOGGER.timerCannotBeAdded(timerEntity);
```

Comment:
you need to add the 'throw' keyword.

Reply:
BTW I had a quick look because the PR title made me curious, and noted the ^^^, but please don't wait on me for approval on this; the EJB experts should do the code review.

Reply:
thx @bstansberry for the feedback. Sry I was testing CI in this case for not breaking anything before asking for Cheng